packages/framework/presence/src/test/notificationsManager.spec.ts

```diff
@@ -3,16 +3,268 @@
  * Licensed under the MIT License.
  */
 
-import type { ISessionClient } from "../index.js";
-import { Notifications } from "../index.js";
+import { strict as assert } from "node:assert";
+
+import { EventAndErrorTrackingLogger } from "@fluidframework/test-utils/internal";
+import { useFakeTimers, type SinonFakeTimers, spy } from "sinon";
+
+import { Notifications, type ISessionClient } from "../index.js";
+import type { IEphemeralRuntime } from "../internalTypes.js";
 import type { IPresence } from "../presence.js";
+import type { createPresenceManager } from "../presenceManager.js";
+
+import { MockEphemeralRuntime } from "./mockEphemeralRuntime.js";
+import { assertFinalExpectations, prepareConnectedPresence } from "./testUtils.js";
+
+/**
+ * A map of presence signals used for testing, indexed by a friendly string.
+ */
+const knownSignals = new Map<string, Parameters<IEphemeralRuntime["submitSignal"]>>([
+	[
+		"PresenceJoin",
+		[
+			"Pres:DatastoreUpdate",
+			{
+				"sendTimestamp": 1010,
+				"avgLatency": 10,
+				"isComplete": true,
+				"data": {
+					"system:presence": {
+						"clientToSessionId": {
+							"client2": { "rev": 0, "timestamp": 1000, "value": "sessionId-2" },
+						},
+					},
+				},
+			},
+		],
+	],
+
+	[
+		"setNewId42",
+		[
+			"Pres:DatastoreUpdate",
+			{
+				"sendTimestamp": 1020,
+				"avgLatency": 10,
+				"data": {
+					"system:presence": {
+						"clientToSessionId": {
+							"client2": { "rev": 0, "timestamp": 1000, "value": "sessionId-2" },
+						},
+					},
+					"n:name:testNotificationWorkspace": {
+						"testEvents": {
+							"sessionId-2": {
+								"rev": 0,
+								"timestamp": 0,
+								"value": { "name": "newId", "args": [42] },
+								"ignoreUnmonitored": true,
+							},
+						},
+					},
+				},
+			},
+		],
+	],
+]);
```

Comment:
Shouldn't need to put this in a Map and deal with `get` that might fail. I think can just use a record. (Maybe your snapshot tool is more compatible with this, but we can update it)
```suggestion
/**
 * A record of presence signals used for testing, indexed by a friendly string.
 */
const knownSignals = {
	"PresenceJoin": [
		"Pres:DatastoreUpdate",
		{
			"sendTimestamp": 1010,
			"avgLatency": 10,
			"isComplete": true,
			"data": {
				"system:presence": {
					"clientToSessionId": {
						"client2": { "rev": 0, "timestamp": 1000, "value": "sessionId-2" },
					},
				},
			},
		},
	],
	"setNewId42": [
		"Pres:DatastoreUpdate",
		{
			"sendTimestamp": 1020,
			"avgLatency": 10,
			"data": {
				"system:presence": {
					"clientToSessionId": {
						"client2": { "rev": 0, "timestamp": 1000, "value": "sessionId-2" },
					},
				},
				"n:name:testNotificationWorkspace": {
					"testEvents": {
						"sessionId-2": {
							"rev": 0,
							"timestamp": 0,
							"value": { "name": "newId", "args": [42] },
							"ignoreUnmonitored": true,
						},
					},
				},
			},
		},
	],
} as const satisfies Record<string, Parameters<IEphemeralRuntime["submitSignal"]>>;
```

Then you can just use `knownSignals.<name>` where needed.

MockEphemeralRuntime ctor will need a parameter update. It is missing readonly qualification:
```diff
-		public readonly signalsExpected: Parameters<IEphemeralRuntime["submitSignal"]>[] = [],
+		public readonly signalsExpected: Readonly<
			Parameters<IEphemeralRuntime["submitSignal"]>
		>[] = [],
```

Reply:
ended up ditching the map completely since it wasn't saving any typing and made the individual tests less clear. Can bring it back if there are more tests. Or refactor the testUtils stuff to make the signals a bit more reusable.